vllm/v1/worker/gpu_model_runner.py

```diff
@@ -14,6 +14,7 @@
 import torch
 import torch.distributed
 import torch.nn as nn
+from torch.autograd.profiler import record_function
```

Comment:
let's use something like this?

```
def record_function_or_nullcontext(name: str, profiling: bool) -> ContextManager:
    if profiling:
        return record_function(name)
    else:
        return contextlib.nullcontext()
```

We can use some engine arg or ENV var to control it.